notification-api.md

```diff
@@ -156,84 +158,82 @@ that should be announced beyond the immediate effect of the primary action.
-not expressly tied to HTML element, but for which other document-centric information can
-be inferred (such as language).
+We provide an API on Documents and Elements to enable posting imperative notifications with 
+independently-managable queues. Additionally, Element-based context should be assumed for
+language.
 
 ```js
-// As the assistive technology to notify the user, given a specific string
-document.ariaNotify( "Selected text is now glowing blue." );
+// Queue a message to the body element's notification queue given the provided string
+document.body.ariaNotify( "Selected text is now glowing blue." );
 ```
+
 A screen reader or other assistive technology tool would speak or show "selected text is now
 glowing blue". For users without assistive technology tools running, nothing would happen.
 The call to the API has no web-observable side effects and its use should not infer that the 
 user is using assistive technology. 
 
-Here the content author provides a label to group a set of related notification about the 
-clipboard.
+The API can also be called on the document. The document also has a single [indpendent] queue
+for notifications:
 
 ```js
-// Use a label to categorize this notification
-document.ariaNotify( "Paste failed.", { label: "clipboard" } );
-document.ariaNotify( "Text copied to clipboard.", { label: "clipboard" } );
+// Use the document's queue.
+document.ariaNotify( "Paste failed." );
+document.ariaNotify( "Text copied to clipboard." );
 ```
 
-The label is used to group or categorize similar notifications. Assistive technology may
-chose to use these labels to provide a filtering mechanism for users.
+Each element (and the document) has a unique queue for notifications. There are some affordances
+for managing these queues and the notifications that go into them (see below). It is implementation-
+dependent how all the various queues are serviced (e.g., round-robin, random selection, FIFO, DOM order),
+though they should all be serviced as part of one 
+[agent](https://html.spec.whatwg.org/multipage/webappapis.html#agents-and-agent-clusters) so as to avoid 
+race-conditions.
 
-Other means of expressing priority and coalescing behavior for similarly-labelled 
-notifications may be defined.
+The API offers two additional options for managing *the current node's* notification queue:
+* `placeInQueue` - how to insert the notification into the queue (at the end, at the start, or to 
+    clear the queue first on insertion).
+* `interruptCurrent` - If an existing notification is being read, is this notification so time-sensitive
+    that it should interterrupt what is currently being read.
 
 ```js
-// Experiment with priority and message combining
-let ariaNotifyOptions = {
-  required: true,
-  priority: "important",
-  label: "formatting commands"
+
+let ariaNotificationOptions = {
+  placeInQueue: "update", // other values "front" & "back" (default value)
```

Comment:
I wonder if "back" should be the default value instead of "update".

Reply:
Oh, that's just an example that is using 'update'. Too subtle that I mentioned 'back' as the default value in the comment?